bin/create_json.py

```diff
@@ -56,29 +61,43 @@
 
             composite_name = table_name.split(".")[0] + "." + variable_name
             if composite_name not in dd_file_json.keys():
-                print("WARNING: variable {} is missing the properties value.".format(composite_name))
+                print("WARNING: variable {} is missing the properties value.".format(
+                    composite_name))
                 if table_name.split(".")[0] in table_name_mapping:
-                    composite_name = table_name_mapping[table_name.split(".")[0]] + "." + variable_name
+                    composite_name = table_name_mapping[table_name.split(
+                        ".")[0]] + "." + variable_name
                     print("trying mapping for " + composite_name)
                     if composite_name not in dd_file_json.keys():
-                        print("WARNING: variable {} is missing the properties value.".format(composite_name))
+                        print("WARNING: variable {} is missing the properties value.".format(
+                            composite_name))
                         continue
                 else:
                     print("WARNING: Missing mapping for " + composite_name)
                     continue
 
-
-            add_codes(dd_file_json[composite_name], yaml_schemas["_terms.yaml"], variable_name, "term", variable_values)
+            if "codes" in dd_file_json[composite_name]:
+                add_codes(dd_file_json[composite_name], yaml_schemas["_terms.yaml"],
+                        variable_name, "term", variable_values)
 
             # TODO need to finish doing the bulk load of the terms before re-enabling this, otherwise the generation of the schema.json will take a very long time
             if "values" in dd_file_json[composite_name]:
+                enum_descriptions = []
                 for value_key, value_value in dd_file_json[composite_name]["values"].items():
-                    add_codes(value_value, yaml_schemas["_terms.yaml"], variable_name, "enumDef", variable_values, value_key)
+                    if "codes" in value_value:
+                        add_codes(value_value, yaml_schemas["_terms.yaml"],
+                                variable_name, "enumDef", variable_values, value_key)
+                    else:
```

Comment:
Why are we adding the description only if the codes are not present?

Reply:
Refactored to make the logic self-contained in a util